controller/controller_server/csi_controller_server.py

```diff
@@ -73,19 +73,16 @@ def CreateVolume(self, request, context):
             with ArrayConnectionManager(user, password, array_addresses) as array_mediator:
                 logger.debug(array_mediator)
 
-                if len(volume_name) > array_mediator.max_vol_name_length:
```

Comment:
then what about xiv and svc now?

Reply:
what do you mean "enough"? what will happen when `volume_name` is larger than 63?

Reply:
volume name comes from pvc name, if the volume name is longer than 63, it means a very long prefix is used. In this case we should raise an error to warn the user instead of cutting the name silently. It is not safe.

Reply:
so where is this error being raised? it seems like now there is no cutting AND no error to warn the user.

Reply:
The error will be raise on storage side (because the name is too long) during creating volume.

Reply:
But error on storage side is too late, you wrote that "we should raise an error to warn the user".
so we can't stop cutting AND leave it without an error to warn the user.

Reply:
ok, raised.

Reply:
Thanks, could you point to where it is raised?

Reply:
So is prefix being ignored for ds8k? we need to think how to communicate this to the user

Reply:
Basically yes, because the prefix will also be hashed. I agree that it is a bit confusing. Maybe we can document that it is not recommended to use prefix in ds8k.

Reply:
"not recommended", or "there is no point" to use it in ds8k?

Reply:
"not recommended", or "there is no point" to use it in ds8k?

Reply:
created pr #128

Reply:
Do you know if there's a way to validate storageclass yamls during `kubectl create/apply` (i.e. before final user pod/volume creation)?

Reply:
yes, thru admission controllers, but it is not csi scope.

Reply:
I see. thanks.

Reply:
@igorgonibm I'm not resolving this only because you opened the conversation.

Reply:
it is storage specific, can not be validated there.

Reply:
Sorry,I don't understand, We agreed that ds8k can use prefix.
And ds8k also have max volume name length limit, I set it to 63
both name and prefix length check will not be added to validate_create_volume_request.
Please explain why validate_publish_volume_request should be moved to mediator classes? I can't see any storage specific logic.

Reply:
The logic is correct. I refer to implementation.
You limit ds8k volume length to 63 when it should be limited only to 16.
a9k and svc should return error when vol name is too long but ds8k do not. So validaton is storage dependent. This is why it should be in mediators.